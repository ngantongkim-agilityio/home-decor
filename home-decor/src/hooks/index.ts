export * from './useAuth';
export * from './useProducts';
export * from './useThemeColor';
