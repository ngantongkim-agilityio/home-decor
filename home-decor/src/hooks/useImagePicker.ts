import { useState } from 'react';
import { Alert } from 'react-native';
import * as ImagePicker from 'expo-image-picker';

export const useImagePicker = () => {
  const [image, setImage] = useState('');

  const pickImageAsync = async () => {
    const result = await ImagePicker.launchImageLibraryAsync({
      mediaTypes: ['images', 'videos'],
      allowsEditing: true,
      aspect: [4, 3],
      quality: 1,
    });

    if (!result.canceled) {
      setImage(result.assets[0].uri);
    } else {
      Alert.alert('You did not select any image.');
    }
  };

  return { image, pickImageAsync };
};
