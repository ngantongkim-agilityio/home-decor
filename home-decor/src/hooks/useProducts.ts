// Libs
import { useInfiniteQuery, useQuery } from '@tanstack/react-query';
import { useShallow } from 'zustand/shallow';

// Constants
import { ROUTES } from '@/constants';

// Services
import { GET } from '@/services';

// Stores
import { authStore } from '@/stores';

// Types
import { IResponseApi } from '@/types';

export const useProducts = () => {
  const useFetchProducts = (initPageParam: number) => {
    const [authKey] = authStore(useShallow((state) => [state.authKey]));

    return useInfiniteQuery({
      queryKey: [ROUTES.PRODUCTS],
      queryFn: ({ pageParam = 1 }) => {
        return GET<IResponseApi<any>>(
          `${ROUTES.PRODUCTS}?page=${pageParam}&per_page=6`,
          {
            headers: {
              'X-Auth-Key': `${authKey?.auth_key}`,
            },
          },
        );
      },
      initialPageParam: initPageParam,
      getNextPageParam: (lastPage) => {
        const { data } = lastPage || {};
        const { listings, page } = data || {};
        const nextPage = page + 1;

        return listings?.length > 0 && listings?.length === 6 ? nextPage : null;
      },
    });
  };

  const useFetchProductDetail = (id: string) => {
    const [authKey] = authStore(useShallow((state) => [state.authKey]));
    console.log('useFetchProductDetail', id, authKey);

    return useQuery({
      queryKey: [id],
      queryFn: () => {
        return GET<IResponseApi<any>>(`${ROUTES.PRODUCTS}/${id}`, {
          headers: {
            'X-Auth-Key': `${authKey?.auth_key}`,
          },
        });
      },
    });
  };
  return { useFetchProducts, useFetchProductDetail };
};
