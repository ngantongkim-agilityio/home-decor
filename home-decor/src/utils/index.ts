export * from './products';
export * from './notifications';
