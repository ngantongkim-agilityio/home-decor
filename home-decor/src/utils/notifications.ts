import * as Notifications from 'expo-notifications';

export const requestPermissionsNotifications = async () => {
  await Notifications.requestPermissionsAsync();
  Notifications.setNotificationHandler({
    handleNotification: async () => ({
      shouldShowAlert: true,
      shouldPlaySound: true,
      shouldSetBadge: false,
    }),
  });
};

export const scheduleNotification = async ({
  id = '',
  title = `You've got a new notification! ðŸ“¬`,
  body = 'Here is the notification message',
  seconds = 5,
}) =>
  await Notifications.scheduleNotificationAsync({
    content: { title, body, data: { url: `/details/${id}` } },
    trigger: {
      seconds,
      repeats: false,
    } as Notifications.TimeIntervalTriggerInput,
  });
