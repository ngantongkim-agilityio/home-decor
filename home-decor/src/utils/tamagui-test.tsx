import { PropsWithChildren, ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import { TamaguiProvider } from 'tamagui';
import { tamaguiConfig } from '@/themes/tamagui.config';

const Providers = ({ children }: PropsWithChildren) => {
  return <TamaguiProvider config={tamaguiConfig}>{children}</TamaguiProvider>;
};

const baseRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react-native';
export { baseRender as render };
