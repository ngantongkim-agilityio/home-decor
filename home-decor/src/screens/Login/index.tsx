import { useCallback, useState } from 'react';
import {
  KeyboardAvoidingView,
  useWindowDimensions,
  ScrollView,
  useColorScheme,
} from 'react-native';
import { Controller, useForm } from 'react-hook-form';
import { useRouter, Link } from 'expo-router';
import { useShallow } from 'zustand/shallow';
import { zodResolver } from '@hookform/resolvers/zod';

// Components
import { Text, Stack, YStack, H2, XStack } from 'tamagui';
import { Button, Input } from '@/components';
import { FacebookIcon, GoogleIcon, BackIcon } from '@/components/icons';

// Types
import { IUserFormInput, ILoginParams, IResponseApi, IUser } from '@/types';

// Hooks
import { useAuth } from '@/hooks';

// Constants
import { ERROR_MESSAGES } from '@/constants';

// Stores
import { authStore, userStore } from '@/stores';

// Schemas
import { LoginSchema } from '@/schemas';

// Themes
import { colors } from '@/themes';

interface IForm {
  email: string;
  password: string;
}

const Login = () => {
  const router = useRouter();
  const colorScheme = useColorScheme();
  const { width, height } = useWindowDimensions();

  const [setAuthKey] = authStore(useShallow((state) => [state.setAuthKey]));
  const [setUser] = userStore(useShallow((state) => [state.setUser]));

  const {
    control,
    handleSubmit,
    formState: { errors },
  } = useForm<IForm>({
    resolver: zodResolver(LoginSchema),
    mode: 'onSubmit',
    defaultValues: {
      email: '',
      password: '',
    },
  });
  const {
    logIn: { mutate, isPending },
  } = useAuth();
  const [error, setError] = useState('');

  const handleBack = useCallback(() => {
    router.back();
  }, [router]);

  const handleLogin = useCallback(
    ({ email, password }: IUserFormInput) => {
      const uuid = `${Date.now()}`;
      const data: ILoginParams<IUserFormInput> = {
        user: {
          email,
          password,
          uuid,
          type: 'customer',
        },
      };

      mutate(data, {
        onSuccess: (data: IResponseApi<ILoginParams<IUser>>) => {
          const { data: authData } = data || {};
          const { user } = authData || {};
          const { key } = user || {};

          !!key && setAuthKey({ ...key, uuid });
          setUser(user);
          router.navigate(`/(tabs)`);
        },
        onError: (error: any) => {
          setError(ERROR_MESSAGES.INVALID_CREDENTIALS);
        },
      });
    },
    [mutate, router, setAuthKey, setUser],
  );

  return (
    <KeyboardAvoidingView behavior="padding" style={{ flex: 1 }}>
      <ScrollView
        showsVerticalScrollIndicator={false}
        contentContainerStyle={{
          flexGrow: 1,
          width,
          justifyContent: 'center',
        }}
      >
        <YStack
          px={24}
          py={40}
          flex={1}
          justify="space-between"
          height={height - 100}
          bg="$bgPrimary"
        >
          <YStack rowGap={40}>
            <XStack items="center" justify="space-between" py={10}>
              <BackIcon onPress={handleBack} />
              <Text color="$primary" fontWeight={600} fontSize={20}>
                Log In
              </Text>
              <Stack width={24} height={24} />
            </XStack>
            <YStack rowGap={11} mb={20}>
              <H2
                fontWeight={600}
                letterSpacing={1}
                fontSize={24}
                color="$textPrimary"
              >
                Welcome
              </H2>
              <Text fontSize={12} color="$textPrimary">
                Please enter your details to proceed.
              </Text>
            </YStack>
            <Stack rowGap={18}>
              <Controller
                name="email"
                control={control}
                render={({ field: { onChange, ...props } }) => {
                  const handleChange = (value: string) => {
                    onChange(value);
                    setError('');
                  };

                  return (
                    <Input
                      id="email"
                      label="Email"
                      placeholder="example@example.com"
                      errorMessage={errors.email?.message}
                      onChangeText={handleChange}
                      {...props}
                    />
                  );
                }}
              />
              <Controller
                name="password"
                control={control}
                render={({ field: { onChange, ...props } }) => {
                  const handleChange = (value: string) => {
                    onChange(value);
                    setError('');
                  };

                  return (
                    <Input
                      id="password"
                      label="Password"
                      placeholder="********"
                      errorMessage={errors.password?.message}
                      secureTextEntry
                      onChangeText={handleChange}
                      {...props}
                    />
                  );
                }}
              />
              {error && (
                <Text color="$error" fontSize={14}>
                  {error}
                </Text>
              )}
            </Stack>
            <YStack items="center" rowGap={18}>
              <Button
                variant="primary"
                width={186}
                isLoading={isPending}
                onPress={handleSubmit(handleLogin)}
              >
                Log In
              </Button>
              <Link href="#">
                <Text fontSize={12} fontWeight={600} color="$textPrimary">
                  Forgot Password?
                </Text>
              </Link>
            </YStack>
          </YStack>
          <YStack items="center" rowGap={16}>
            <Text fontSize={12} fontWeight={300} color="$textPrimary">
              or sign up with
            </Text>
            <XStack columnGap={20} mb={10}>
              <FacebookIcon
                color={colorScheme === 'light' ? colors.bgDark : colors.primary}
                onPress={() => {}}
              />
              <GoogleIcon
                color={colorScheme === 'light' ? colors.bgDark : colors.primary}
                onPress={() => {}}
              />
            </XStack>
            <Text fontSize={12} fontWeight={300} color="$textPrimary">
              {`Donâ€™t have an account? `}
              <Link href={'/(auth)/(signup)'}>
                <Text ml={4} fontSize={12} color="$secondary">
                  Sign Up
                </Text>
              </Link>
            </Text>
          </YStack>
        </YStack>
      </ScrollView>
    </KeyboardAvoidingView>
  );
};

export default Login;
