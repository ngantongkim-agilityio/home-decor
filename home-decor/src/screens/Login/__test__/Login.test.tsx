import { fireEvent, render, waitFor } from '@/utils/tamagui-test';
import { useAuth } from '@/hooks';
import Login from '..';

jest.mock('@/hooks', () => ({
  useAuth: jest.fn(),
}));

export const mockLoginForm = { email: 'test@gmail.com', password: 'Test123' };

describe('Login Screen', () => {
  const mockLogin = jest.fn();

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Should render to match snapshot', () => {
    (useAuth as jest.Mock).mockReturnValue({
      logIn: { mutate: mockLogin },
    });
    const component = render(<Login />);

    expect(component).toMatchSnapshot();
  });

  it('Should render the login form correctly', () => {
    (useAuth as jest.Mock).mockReturnValue({
      logIn: { mutate: mockLogin },
    });
    const { getByText, getAllByText } = render(<Login />);

    expect(getAllByText(/Log In/i)).toBeTruthy();
    expect(getByText(/Welcome/i)).toBeTruthy();
    expect(getByText(/Please enter your details to proceed./i)).toBeTruthy();
    expect(getByText(/Donâ€™t have an account?/i)).toBeTruthy();
  });

  it('Should call login function when form is submitted', async () => {
    (useAuth as jest.Mock).mockReturnValue({
      logIn: { mutate: mockLogin },
    });
    const { getByLabelText, getAllByText } = render(<Login />);

    const emailInput = getByLabelText(/Email/i);
    const passwordInput = getByLabelText(/Password/i);
    const saveButton = getAllByText('Log In')[1];

    fireEvent.changeText(emailInput, mockLoginForm.email);
    fireEvent.changeText(passwordInput, mockLoginForm.password);
    fireEvent.press(saveButton);

    await waitFor(() => {
      expect(mockLogin).toHaveBeenCalledTimes(1);
    });
  });
});
