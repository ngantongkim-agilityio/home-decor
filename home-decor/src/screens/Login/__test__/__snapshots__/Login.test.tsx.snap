// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Screen Should render to match snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      {
        "paddingBottom": 0,
      },
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      {
        "flexGrow": 1,
        "justifyContent": "center",
        "width": 750,
      }
    }
    showsVerticalScrollIndicator={false}
  >
    <View>
      <View
        style={
          {
            "backgroundColor": {
              "dynamic": {
                "dark": "#36312f",
                "light": "#ffffff",
              },
            },
            "flex": 1,
            "flexDirection": "column",
            "height": 1234,
            "justifyContent": "space-between",
            "paddingBottom": 40,
            "paddingLeft": 24,
            "paddingRight": 24,
            "paddingTop": 40,
          }
        }
      >
        <View
          style={[]}
        >
          <View
            style={[]}
          >
            <RNSVGSvgView
              bbHeight={19}
              bbWidth={22}
              fill="none"
              focusable={false}
              height={19}
              onPress={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              responsible={true}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "flex": 0,
                    "height": 19,
                    "width": 22,
                  },
                ]
              }
              width={22}
            >
              <RNSVGGroup
                fill={null}
                propList={
                  [
                    "fill",
                  ]
                }
              >
                <RNSVGPath
                  d="M11.909 1 1 9.497m0 0L11.909 18M1 9.497h19.62"
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "stroke",
                      "strokeWidth",
                      "strokeLinecap",
                      "strokeLinejoin",
                    ]
                  }
                  stroke={
                    {
                      "payload": 4283123012,
                      "type": 0,
                    }
                  }
                  strokeLinecap={1}
                  strokeLinejoin={1}
                  strokeWidth={1.29}
                />
              </RNSVGGroup>
            </RNSVGSvgView>
            <Text
              style={[]}
            >
              Log In
            </Text>
            <View
              style={[]}
            />
          </View>
          <View
            style={[]}
          >
            <Text
              style={[]}
            >
              Welcome
            </Text>
            <Text
              style={[]}
            >
              Please enter your details to proceed.
            </Text>
          </View>
          <View
            style={[]}
          >
            <View
              style={[]}
            >
              <Text
                htmlFor="email"
                id=":r3:"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "color": {
                      "dynamic": {
                        "dark": "#ffffff",
                        "light": "#363130",
                      },
                    },
                    "display": "flex",
                    "fontFamily": "PoppinsRegular",
                    "fontSize": 14,
                    "lineHeight": 20,
                    "userSelect": "none",
                  }
                }
                suppressHighlighting={true}
              >
                Email
              </Text>
              <TextInput
                aria-label="Email"
                focusVisibleStyle={{}}
                focusable={true}
                id="email"
                name="email"
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="example@example.com"
                placeholderTextColor={
                  {
                    "dynamic": {
                      "dark": "#c0c0c0",
                      "light": "#c0c0c0",
                    },
                  }
                }
                readOnly={false}
                style={
                  {
                    "backgroundColor": {
                      "dynamic": {
                        "dark": "#f0e1d2",
                        "light": "#f0e1d2",
                      },
                    },
                    "borderBottomLeftRadius": 50,
                    "borderBottomRightRadius": 50,
                    "borderBottomWidth": 0,
                    "borderLeftWidth": 0,
                    "borderRightWidth": 0,
                    "borderStyle": "solid",
                    "borderTopLeftRadius": 50,
                    "borderTopRightRadius": 50,
                    "borderTopWidth": 0,
                    "fontFamily": "PoppinsRegular",
                    "fontSize": 14,
                    "height": 41,
                    "minWidth": 0,
                    "paddingBottom": 10,
                    "paddingLeft": 10,
                    "paddingRight": 10,
                    "paddingTop": 10,
                  }
                }
                value=""
              />
            </View>
            <View
              style={[]}
            >
              <Text
                htmlFor="password"
                id=":r9:"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "color": {
                      "dynamic": {
                        "dark": "#ffffff",
                        "light": "#363130",
                      },
                    },
                    "display": "flex",
                    "fontFamily": "PoppinsRegular",
                    "fontSize": 14,
                    "lineHeight": 20,
                    "userSelect": "none",
                  }
                }
                suppressHighlighting={true}
              >
                Password
              </Text>
              <TextInput
                aria-label="Password"
                focusVisibleStyle={{}}
                focusable={true}
                id="password"
                name="password"
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="********"
                placeholderTextColor={
                  {
                    "dynamic": {
                      "dark": "#c0c0c0",
                      "light": "#c0c0c0",
                    },
                  }
                }
                readOnly={false}
                secureTextEntry={true}
                style={
                  {
                    "backgroundColor": {
                      "dynamic": {
                        "dark": "#f0e1d2",
                        "light": "#f0e1d2",
                      },
                    },
                    "borderBottomLeftRadius": 50,
                    "borderBottomRightRadius": 50,
                    "borderBottomWidth": 0,
                    "borderLeftWidth": 0,
                    "borderRightWidth": 0,
                    "borderStyle": "solid",
                    "borderTopLeftRadius": 50,
                    "borderTopRightRadius": 50,
                    "borderTopWidth": 0,
                    "fontFamily": "PoppinsRegular",
                    "fontSize": 14,
                    "height": 41,
                    "minWidth": 0,
                    "paddingBottom": 10,
                    "paddingLeft": 10,
                    "paddingRight": 10,
                    "paddingTop": 10,
                  }
                }
                value=""
              />
            </View>
          </View>
          <View
            style={[]}
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": {
                      "dynamic": {
                        "dark": "#f4b5a4",
                        "light": "#f4b5a4",
                      },
                    },
                    "borderBottomLeftRadius": 50,
                    "borderBottomRightRadius": 50,
                    "borderBottomWidth": 0,
                    "borderLeftWidth": 0,
                    "borderRightWidth": 0,
                    "borderStyle": "solid",
                    "borderTopLeftRadius": 50,
                    "borderTopRightRadius": 50,
                    "borderTopWidth": 0,
                    "flexDirection": "row",
                    "gap": 10,
                    "height": 41,
                    "justifyContent": "center",
                    "opacity": 1,
                    "width": 186,
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": {
                        "dynamic": {
                          "dark": "#cc7862",
                          "light": "#cc7862",
                        },
                      },
                      "fontFamily": "PoppinsBold",
                      "fontSize": 15,
                      "lineHeight": 21,
                      "textDecorationLine": "none",
                    }
                  }
                  suppressHighlighting={true}
                >
                  Log In
                </Text>
              </View>
            </View>
            <Text
              href="#"
              onPress={[Function]}
              role="link"
            >
              <Text
                style={[]}
              >
                Forgot Password?
              </Text>
            </Text>
          </View>
        </View>
        <View
          style={[]}
        >
          <Text
            style={[]}
          >
            or sign up with
          </Text>
          <View
            style={[]}
          >
            <RNSVGSvgView
              bbHeight={34}
              bbWidth={33}
              fill="none"
              focusable={false}
              height={34}
              onPress={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              responsible={true}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "flex": 0,
                    "height": 34,
                    "width": 33,
                  },
                ]
              }
              width={33}
            >
              <RNSVGGroup
                fill={null}
                propList={
                  [
                    "fill",
                  ]
                }
              >
                <RNSVGCircle
                  cx={16.5}
                  cy={17.383}
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "stroke",
                    ]
                  }
                  r={16}
                  stroke={
                    {
                      "payload": 4283123012,
                      "type": 0,
                    }
                  }
                />
                <RNSVGPath
                  d="M17.815 27.88v-5.886h3.042a.056.056 0 0 0 .056-.055v-3.76a.056.056 0 0 0-.056-.055h-3.039v-1.722c0-.305.122-.597.337-.813.215-.216.506-.338.81-.339h1.944v-3.495a.376.376 0 0 0-.364-.359h-2.884a3.827 3.827 0 0 0-2.712 1.128 3.859 3.859 0 0 0-1.124 2.723s-.039.56-.068 2.873h-2.95a.111.111 0 0 0-.112.112v3.647a.112.112 0 0 0 .111.112h2.928v5.596m.016.01a10.49 10.49 0 0 1-5.89-4.18 10.565 10.565 0 0 1 1.154-13.379 10.493 10.493 0 0 1 6.29-3.085 10.45 10.45 0 0 1 8.188 2.602 10.53 10.53 0 0 1 2.593 3.558A10.57 10.57 0 0 1 27 17.426a10.557 10.557 0 0 1-2.62 6.97 10.477 10.477 0 0 1-6.559 3.487"
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "stroke",
                      "strokeLinecap",
                      "strokeLinejoin",
                    ]
                  }
                  stroke={
                    {
                      "payload": 4283123012,
                      "type": 0,
                    }
                  }
                  strokeLinecap={1}
                  strokeLinejoin={1}
                />
              </RNSVGGroup>
            </RNSVGSvgView>
            <RNSVGSvgView
              bbHeight={34}
              bbWidth={34}
              fill="none"
              focusable={false}
              height={34}
              onPress={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              responsible={true}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "flex": 0,
                    "height": 34,
                    "width": 34,
                  },
                ]
              }
              width={34}
            >
              <RNSVGGroup
                fill={null}
                propList={
                  [
                    "fill",
                  ]
                }
              >
                <RNSVGCircle
                  cx={17.311}
                  cy={17.383}
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "stroke",
                    ]
                  }
                  r={16}
                  stroke={
                    {
                      "payload": 4283123012,
                      "type": 0,
                    }
                  }
                />
                <RNSVGPath
                  d="M22.586 11.074a8.195 8.195 0 1 0 2.643 6.028h-8.195"
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "stroke",
                      "strokeLinecap",
                      "strokeLinejoin",
                    ]
                  }
                  stroke={
                    {
                      "payload": 4283123012,
                      "type": 0,
                    }
                  }
                  strokeLinecap={1}
                  strokeLinejoin={1}
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
          <Text
            style={[]}
          >
            Don’t have an account? 
            <Text
              href="/"
              onPress={[Function]}
              role="link"
            >
              <Text
                style={[]}
              >
                Sign Up
              </Text>
            </Text>
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
